hash/hash_comparator.go

```diff
@@ -200,7 +240,10 @@ func CompareMD5(_ context.Context, password []byte, hash []byte) error {
 }
 
 var (
+	isMD5CryptHash       = regexp.MustCompile(`^\$1\$`)
```

Comment:
Can we use a more descriptive label like "md5-crypt" here instead of just `1`?

Reply:
Sure, I considered doing that (something like `md5-crypt`). Still, I decided against it as `bcrypt` already uses their Modular Crypt Identifier (https://manpages.debian.org/bullseye/libcrypt-dev/crypt.5.en.html#bcrypt), and `$1$` is the standardized one for md5-crypt (as is `$5$` and `$6$` for sha256crypt and sha512crypt respectively.)

Reply:
fixed in [216f1db](https://github.com/ory/kratos/pull/3303/commits/216f1dbbd4a8febf3e3550c6765057df4a6a8d4d)